eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ServiceClientBuilderCheck.java

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.tools.checkstyle.checks;
+
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import java.util.Stack;
+
+/**
+ * The @ServiceClientBuilder class should have the following rules:
+ *  1) All service client builder should be named <ServiceName>ClientBuilder and annotated with @ServiceClientBuilder.
+ *  2) Has a method 'buildClient()' to build a synchronous client,
+ *  3) Has a method 'buildAsyncClient()' to build an asynchronous client
```

Comment:
Not all builders need to have both methods. For example, Event Hubs is only async at the moment. There should be no other methods starting with 'build' other than these two.

Reply:
Does every @ServiceClientBuilder class should have one of buildClient or buildAsyncClient method? or it could be none.

Reply:
Probably the best answer is that there must be at least one method named build*Client(), where * could be any string zero or more characters long. That is, buildClient() is valid, so is buildAsyncClient(), buildBlobClient(), buildBlobAsyncClient(), etc. So, we should expect at least one of these.

Additionally, I think for now we could also have the rule enforce the requirement that any method that starts with `build` must also end with `Client`.